aiohttp/client_proto.py

```diff
@@ -275,14 +275,19 @@ def data_received(self, data: bytes) -> None:
                 # parse http messages
                 try:
                     messages, upgraded, tail = self._parser.feed_data(data)
-                except BaseException as underlying_exc:
+                except HttpProcessingError as underlying_exc:
                     if self.transport is not None:
                         # connection.release() could be called BEFORE
                         # data_received(), the transport is already
                         # closed in this case
                         self.transport.close()
```

Comment:
I feel like that `BaseException` was here to make sure to close transport in case of unexpected crashes, right? Is it possible for some non-matching exception to sneak in through this code path https://github.com/aio-libs/aiohttp/blob/bf022b3/aiohttp/_http_parser.pyx#L551C20-L551C36 ?

Reply:
Hmm, it does catch BaseException in the parser, but I don't see any reason we'd have an exception there...

Reply:
It feels a little awkward, but guess we'll keep it like that for now.

Reply:
We're obviously missing coverage now, because there's no known situation where this could happen though..